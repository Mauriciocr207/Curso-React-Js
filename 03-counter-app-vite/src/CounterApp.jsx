import { number } from 'prop-types'
import { useState } from 'react';

export default function CounterApp({value}) {
    const [count, setCount] = useState(value);

    function handleIncrement() {
        setCount(p => ++p);
    }

    function handleDecrement() {
        setCount(p => --p);
    }

    function handleReset() {
        setCount(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2 data-testid="CounterApp.count">{count}</h2>
            <button data-testid="CounterApp.inc" onClick={handleIncrement}>+</button>
            <button data-testid="CounterApp.dec" onClick={handleDecrement}>-</button>
            <button data-testid="CounterApp.res" onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: number.isRequired,
}