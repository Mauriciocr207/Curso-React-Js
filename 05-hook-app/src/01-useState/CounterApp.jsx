import { useState } from "react";

export default function CounterApp() {
  const [{ counter1, counter2, counter3 }, setCounter] = useState({
    counter1: 0,
    counter2: 0,
    counter3: 0,
  });

  const onClick = () =>
    setCounter(({ counter1, ...state }) => ({
      counter1: counter1 + 1,
      ...state,
    }));

  return (
    <>
      <h2 className="text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl">
        Counter App
      </h2>
      <p className="text-lg mt-10">Counter1: {counter1}</p>
      <p className="text-lg mt-10">Counter1: {counter2}</p>
      <p className="text-lg mt-10">Counter1: {counter3}</p>
      <div className="mt-10 flex items-center justify-center gap-x-6">
        <button
          onClick={onClick}
          className="rounded-md bg-gray-900 px-3.5 py-2.5 text-xl font-semibold text-white shadow-sm hover:bg-gray-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-white"
        >
          +
        </button>
      </div>
    </>
  );
}
