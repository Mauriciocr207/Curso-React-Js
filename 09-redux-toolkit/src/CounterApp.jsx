import reactLogo from './assets/react.svg'
import { useDispatch, useSelector } from 'react-redux'
import { decrement, increment, incrementByAmount } from './features/counter/counterSlice';

export default function CounterApp() {
  const count = useSelector(state => state.counter.value);
  const dispatch = useDispatch();

  return (
    <>
      <div>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Your Counter App</h1>
      <p>count is {count} </p>
      <div className="card" style={{display: "flex", gap: "2rem"}}>
        <button onClick={() => dispatch(increment())}>
          increment
        </button>
        <button onClick={() => dispatch(decrement())}>
          decrement
        </button>
        <button onClick={() => dispatch(incrementByAmount(5))}>
          increment by 5
        </button>
      </div>
    </>
  )
}
