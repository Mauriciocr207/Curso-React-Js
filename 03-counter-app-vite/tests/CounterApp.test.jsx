import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import CounterApp from "../src/CounterApp"

describe('CounterApp test', () => {
    const initialValue = 100;

    test('valid snapshot', () => {
        const {container} = render(<CounterApp value={initialValue}/>);
        expect(container).toMatchSnapshot();
    })

    test('initial value is 100', () => {
        render(<CounterApp value={initialValue}/>);
        const count = screen.getByTestId('CounterApp.count');
        expect(count).toBeTruthy();
        expect(+count.textContent).toBe(100);
    })

    test('increment value', async () => {
        const user = userEvent.setup();
        render(<CounterApp value={initialValue}/>);
        const incrementBtn = screen.getByTestId('CounterApp.inc');
        const count = screen.getByTestId('CounterApp.count');

        await user.click(incrementBtn);

        expect(+count.textContent).toBe(101);
    })
    
    test('decrement value', async () => {
        const user = userEvent.setup();
        render(<CounterApp value={initialValue}/>);
        const incrementBtn = screen.getByTestId('CounterApp.dec');
        const count = screen.getByTestId('CounterApp.count');

        await user.click(incrementBtn);

        expect(+count.textContent).toBe(99);
    })
    
    test('reset value', async () => {
        const user = userEvent.setup();
        render(<CounterApp value={initialValue}/>);
        const incrementBtn = screen.getByTestId('CounterApp.res');
        const count = screen.getByTestId('CounterApp.count');

        await user.click(incrementBtn);

        expect(+count.textContent).toBe(100);
    })
})