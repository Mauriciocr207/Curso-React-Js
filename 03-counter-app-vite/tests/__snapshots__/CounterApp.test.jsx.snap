// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp test valid snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="CounterApp.count"
  >
    100
  </h2>
  <button
    data-testid="CounterApp.inc"
  >
    +
  </button>
  <button
    data-testid="CounterApp.dec"
  >
    -
  </button>
  <button
    data-testid="CounterApp.res"
  >
    Reset
  </button>
</div>
`;
